client-mr/core/src/main/java/org/apache/hadoop/mapreduce/RssMRUtils.java

```diff
@@ -44,37 +44,40 @@ public class RssMRUtils {
 
   private static final Logger LOG = LoggerFactory.getLogger(RssMRUtils.class);
   private static final BlockIdLayout LAYOUT = BlockIdLayout.DEFAULT;
-  private static final int MAX_ATTEMPT_LENGTH = 6;
+  private static final int MAX_ATTEMPT_LENGTH = 4;
   private static final int MAX_ATTEMPT_ID = (1 << MAX_ATTEMPT_LENGTH) - 1;
-  private static final int MAX_SEQUENCE_NO =
-      (1 << (LAYOUT.sequenceNoBits - MAX_ATTEMPT_LENGTH)) - 1;
+  private static final int MAX_TASK_LENGTH = LAYOUT.taskAttemptIdBits - MAX_ATTEMPT_LENGTH;
+  private static final int MAX_TASK_ID = (1 << MAX_TASK_LENGTH) - 1;
 
-  // Class TaskAttemptId have two field id and mapId, rss taskAttemptID have 21 bits,
-  // mapId is 19 bits, id is 2 bits. MR have a trick logic, taskAttemptId will increase
-  // 1000 * (appAttemptId - 1), so we will decrease it.
+  // Class TaskAttemptId have two field id and mapId. MR have a trick logic, taskAttemptId will
+  // increase 1000 * (appAttemptId - 1), so we will decrease it.
   public static long convertTaskAttemptIdToLong(TaskAttemptID taskAttemptID, int appAttemptId) {
```

Comment:
I'd suggest to improve the naming.

We generate a taskAttemptId to be used in blockId / RSS, which should be called `RssTaskAttemptId`:

```suggestion
  public static int createRssTaskAttemptId(TaskAttemptID taskAttemptID, int appAttemptId) {
```

The reverse operation is getting a MR / TEZ taskAttemptId from the `RssTaskAttemptId`:

We already have the reverse method that follows that naming convention:

    public static TaskAttemptID createMRTaskAttemptId(...)

With that naming convention, it is clear which type of task attempt id is meant in each of these methods.